controllers/gslb_controller_reconciliation.go

```diff
@@ -197,29 +154,27 @@ func (r *GslbReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 	gslb.Status.ServiceHealth = serviceHealth
 
 	// == external-dns dnsendpoints CRs ==
-	dnsEndpoint, err := r.gslbDNSEndpoint(gslb)
+	_, s := r.Tracer.Start(context.Background(), "gslbDNSEndpoint")
+	epProvider := k8gbendpoint.NewApplicationDNSEndpoint(context.TODO(), r.Client, r.Config, gslb, log, r.updateRuntimeStatus)
```

Comment:
Is it intentional usage of TODO context? Something to consider https://dev.to/zakariachahboun/are-you-confused-about-contexttodo-vs-contextbackground-heres-what-you-need-to-know-2h46

Reply:
thanks @ytsarev , I’ll delete it in the following PR.

Just for info, here’s why it’s there: I deliberately insert this block of code so I can debug k8gb from my own IDE against local test clusters.

Example: k8gb is running on the EU cluster. The k8gb instance for the US cluster is running in the debugger, and I’m running Terratests in a second debugger instance. Very useful.

Reply:
That’s a good question. I had previously thought about whether there should rather be a single context within one reconciler, so that in case of a hang, the loop would get cancelled.

This was meant to be just a context within a single trace call.